File: leetcode-submissions/763_Partition_Labels.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        unordered_map<char, int >mp; // it stores character to last index 
        for(int i=0;i<s.length();i++){
            mp[s[i]]= i ;
        }
        int maxIndex = -1;
        int lastIndex = 0 ;
        vector<int>ans;
        for(int i=0;i<s.length();i++){
            maxIndex = max(mp[s[i]], maxIndex);
            if(maxIndex==i){
                // partition 
                ans.push_back(i-lastIndex+(lastIndex==0));
                lastIndex= i;
                maxIndex= -1;
            }

        }
        return ans ;
    }
};

File: leetcode-submissions/763_Partition_Labels.ts
function partitionLabels(s: string): number[] {
    let mp:Map<string, number> = new Map();
    let lastUsedIndex :number=0 , maxIndex :number =-1;
    const n :number = s.length, ans :number[] =[];
    for(let i:number =0;i<n;i++){
        mp.set(s[i], i);
    }
    for(let i:number =0;i<n;i++){
        maxIndex = Math.max(maxIndex , mp.get(s[i]));
        if(maxIndex==i){
            ans.push(i-lastUsedIndex+((ans.length===0)?1:0));
            lastUsedIndex = i ;
            maxIndex = -1;
        }
    }
    return ans ;
};